### DISCUSSION TITLE
opinion

### DISCUSSION THREAD

**Discussion by nadongjin** [leondgarse/Keras_insightface#74] [Q&A]
I ask for your opinion.
If the dataset is too large, there is a low probability that a similar person in another class will enter the batch.
So, all images for each class are vector transformed using a pre-trained engine.
Then, only the most similar people in different classes are constantly grouped into batches to learn.
Personally, I think this can reduce learning time and give more distinction to similar people.
how do you think?

**Comment by leondgarse** [comment]
I think that's what `Triplet` doing, the `offline triplet mining`. Technically, it's digging positive and negative pairs in a large data batch, or even the whole dataset, then feed the hardest ones for model training. I'm testing `offline triplet mining` + `triplet loss with arcface margin` for performance and if any improve, as current `triplet loss` digging only in the `batch_size` data is not working well.
